core/services/log/broadcaster_test.go

```diff
@@ -167,24 +165,24 @@ func TestBroadcaster_BackfillOnNodeStartAndOnReplay(t *testing.T) {
 	// the first backfill should use the height of last head saved to the db,
 	// minus maxNumConfirmations of subscribers and minus blockBackfillDepth
 	mockEth.checkFilterLogs = func(fromBlock int64, toBlock int64) {
-		times := atomic.LoadInt64(backfillCountPtr)
+		times := backfillCount.Load()
 		if times == 0 {
 			require.Equal(t, lastStoredBlockHeight-maxNumConfirmations-int64(blockBackfillDepth), fromBlock)
 		} else if times == 1 {
 			require.Equal(t, replayFrom, fromBlock)
 		}
 
-		atomic.StoreInt64(backfillCountPtr, times+1)
+		backfillCount.Store(times + 1)
```

Comment:
Do these `checkFilterLogs` funcs ever execute concurrently? If so, the ones incrementing via a `Store` call at the end will race.

Reply:
The first CI failure was the related test `TestBroadcaster_BackfillALargeNumberOfLogs` but it passed on a re-run, which suggests this is a race - fix incoming.

The second CI failure was an unrelated test `TestIntegration_DirectRequest`.

Reply:
so the TestBroadcaster_BackfillALargeNumberOfLogs i just fixed in https://github.com/smartcontractkit/chainlink/pull/4950 
and DirectRequest is actually an indication of a non-test issue

Reply:
`TestIntegration_DirectRequest` is being addressed here: https://github.com/smartcontractkit/chainlink/pull/4956 and CI ran `TestBroadcaster_BackfillALargeNumberOfLogs` successfully twice now.